packages/assets-controllers/src/NftController.ts

```diff
@@ -1383,16 +1447,26 @@ export class NftController extends BaseController<
     tokenId: string,
     {
       nftMetadata,
-      userAddress = this.#selectedAddress,
+      userAddress,
       source = Source.Custom,
       networkClientId,
     }: {
       nftMetadata?: NftMetadata;
       userAddress?: string;
       source?: Source;
       networkClientId?: NetworkClientId;
-    } = { userAddress: this.#selectedAddress },
+    } = {},
   ) {
+    const selectedAccount = this.messagingSystem.call(
+      'AccountsController:getAccount',
+      this.#selectedAccountId,
+    );
+    // Previously selectedAddress could be an empty string. This is to preserve the behaviour
+    userAddress = userAddress || selectedAccount?.address || '';
+    if (!userAddress) {
```

Comment:
Why do we check for this one here? The original behavior was not, right?

Reply:
It would still work like normal without it, but the behaviour with an empty address is adding the nft with an empty string as the key. It just bloats the state.